### DISCUSSION TITLE
huawei p40 with react-native-firebase

### DISCUSSION THREAD

**Discussion by edrickkochla** [invertase/react-native-firebase#4901] [Q&A] [answered]
Can react-native-firebase be implemented on latest huawei phones, for example, huawei p40?
Our current react-native-firebase version is 6.3.4. According to the installation process for react native firebase it requires the google services plugin (apply plugin: 'com.google.gms.google-services', according to documentation here --> https://rnfirebase.io/) but new huawei phones do not support google services. The only solution I can think of is to implement firebase REST api. Is there another workaround? Is there a newer version which will work on new huawei phones? I understand firebase authentication does not depend on google services anymore as many others services also do not, but the installation process for this library still require google services and therefore it seems that this library will not work on huawei p40. Any solutions or contributions will be highly appreciated.

**Comment by mikehardy** [comment, marked as answer]
This is the canonical documentation: https://firebase.google.com/docs/android/android-play-services

It will be based on current versions though, and 6.3.4 is definitely not current, so if you make decisions based on that document you'll want to be up to date here (10.7.x at the moment) so you get latest firebase-android-sdk (26.4.0 at time of me typing this).

Then I would say you should try it and see

It is also important to understand what the android google-services plugin does, it is not "play services", it is "google services" - it's just a convenient way to get the contents of google-services.json (your firebase project definition, in other words) injected into the app. It actually has no bearing on Play Services.

**Reply by edrickkochla** [reply]
awesome! Much appreciated
